### PR TITLE
[Feature] Migrate circleci to action

### PR DESCRIPTION
**Pull request by Atralupus**
Related to https://github.com/planetarium/9c-launcher/issues/2132

Migrate CircleCI to GitHub Actions for later enhancements in the launcher's CI/CD process
please check this action (https://github.com/planetarium/9c-launcher/actions/runs/4677166441)

### CODE REVIEW

File: .github/actions/yarn-install/action.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: yarn-install
+runs:
+  using: composite
+  steps:
+    - name: restore_cache
+      uses: actions/cache@v3.3.1
+      with:
+        key: yarn3-packages-{{ arch }}-{{ checksum "yarn.lock" }}
+        path: ".yarn/cache"
+        restore-keys: yarn3-packages-{{ arch }}-{{ checksum "yarn.lock" }}
```
Comment (moreal) [inline .github/actions/yarn-install/action.yaml:10]: If it wanted to use [GitHub Actions Context](https://docs.github.com/en/actions/learn-github-actions/contexts) feature, it should be fixed manually. I didn't test it but it should be like:

```suggestion
        key: yarn3-packages-${{ runner.arch }}-${{ hashFiles("yarn.lock") }}
        path: ".yarn/cache"
        restore-keys: yarn3-packages-${{ runner.arch }}-${{ hashFiles("yarn.lock") }}
```

- `runner.arch`: https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context
- `hashFiles(...)`: https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles

File: .github/actions/yarn-install/action.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    - run: yarn install --immutable
+      env:
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+      shell: bash
+
+    - name: save_cache
+      uses: actions/cache@v3.3.1
+      with:
+        path: ".yarn/cache"
+        key: yarn3-packages-{{ arch }}-{{ checksum "yarn.lock" }}
```
Comment (moreal) [inline .github/actions/yarn-install/action.yaml:21]: See the above comment please:
```suggestion
        key: yarn3-packages-${{ runner.arch }}-${{ hashFiles("yarn.lock") }}
```

File: .github/actions/yarn-install/action.yaml (hunk 1 above)
Comment (Atralupus) [reply .github/actions/yarn-install/action.yaml]: good point, thanks

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      - name: Setup OS variables
+        run: bash scripts/configure-artifact.sh ${{ matrix.targetPlatform }}
+
+      - name: Copy config.json from the last release
+        shell: bash
+        run: |
+          mkdir pack-dist
+          set -ex
+          mkdir -p dist
+          curl -o "dist/config.json" 'https://release.nine-chronicles.com/9c-launcher-config.json'
```
Comment (maxswjeon) [inline .github/workflows/build.yml:48]: I think the `rc-v*` builds should get config.json from `internal/` folder

**Review by maxswjeon — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (maxswjeon) [inline .github/workflows/build.yml:48]: I think the `rc-v*` builds should get config.json from `internal/` folder

Comment (maxswjeon) [reply .github/workflows/build.yml:48]: My mistake.

**Review by moreal — APPROVED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (Atralupus) [reply .github/workflows/build.yml:48]: This action simply migrates the existing CircleCI action. I will perform refactoring later.

**Pull request merged**